src/main/java/ui/components/pickers/RepositoryPickerDialog.java

```diff
@@ -0,0 +1,149 @@
+package ui.components.pickers;
+
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.scene.Node;
+import javafx.scene.control.*;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.VBox;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import ui.IdGenerator;
+import util.Utility;
+
+import java.util.*;
+
+/**
+ * Serves as a presenter that also helps the user by suggesting repositories stored in the disk
+ * which are related to user input.
+ */
+public class RepositoryPickerDialog extends Dialog<String> {
+
+    private static final String REPO_PICKER_TITLE = "Pick another repository";
+    private static final int DEFAULT_SUGGESTED_REPO_LIST_HEIGHT = 350;
+    private static final int DEFAULT_SUGGESTED_REPO_LIST_WIDTH = 350;
+    private static final int SPACING_BETWEEN_NODES = 10;
+    private static final Insets DEFAULT_PADDING = new Insets(10);
+    private static final Insets DEFAULT_LABEL_MARGIN = new Insets(2);
+
+    private VBox mainLayout;
+    private VBox suggestedRepositoryList;
+    private TextField userInputTextField;
+    private RepositoryPickerState state;
+
+    public RepositoryPickerDialog(Set<String> storedRepos, Stage stage) {
+        initUi(stage, storedRepos);
+    }
+
+    private void initUi(Stage stage, Set<String> storedRepos) {
+        state = new RepositoryPickerState(storedRepos);
+
+        initialiseDialog(stage);
+        createButtons();
+        initialiseDefaultValues();
+
+        Platform.runLater(() -> userInputTextField.requestFocus());
+    }
+
+    private void initialiseDefaultValues() {
```

Comment:
Not very clear what are the effects of this method i.e. what are the `default values`, so could be nice to have a quick comments about this.